plugins/inventory/terraform_state.py

```diff
@@ -51,6 +51,30 @@
     description:
       - The path of a terraform binary to use.
     type: path
+  provider_mapping:
+    description:
+      - List of custom provider mappings.
+    type: list
+    elements: dict
+    default: []
+    required: False
+    suboptions:
+      name:
+        description:
+          - Arbitrary name
+        type: str
+        required: True
```

Comment:
I added this only because the 'ProviderMapping' is a map that needs a key. But I'm going to remove it.

```
ProvidersMapping = {
    "aws": TerraformProviderInstance(provider_name="registry.terraform.io/hashicorp/aws", types=["aws_instance"]),
    "azurerm": TerraformProviderInstance(
        provider_name="registry.terraform.io/hashicorp/azurerm",
        types=["azurerm_virtual_machine", "azurerm_linux_virtual_machine", "azurerm_windows_virtual_machine"],
    ),
    "google": TerraformProviderInstance(
        provider_name="registry.terraform.io/hashicorp/google", types=["google_compute_instance"]
    ),
}
```